src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -139,6 +141,7 @@ private struct ComInterfaceInstance
         /// this <see cref="ComWrappers" /> instance, the previously created COM interface will be returned.
         /// If not, a new one will be created.
         /// </remarks>
+        [SupportedOSPlatform("windows")]
```

Comment:
Can we just mark the whole type as windows specific?

Reply:
Seems reasonable to me

Reply:
done. 

BTW how should we handle nested types? If `class B` is nested within `class A` which is Windows-specific, should `B` be Windows-specific OOTB as well?

```cs
[SupportedOSPlatform("windows")]
public class A
{
    public class B { }
}
```

/cc @buyaa-n

Reply:
>  If class B is nested within class A which is Windows-specific, should B be Windows-specific OOTB as well?

Yes, I would assume so. The same with any other scope - if a class is marked OS-specific, all the methods in it are OS-specific. If an assembly is marked OS-specific, all the types in it are OS-specific.